### PR TITLE
Interactions API Endpoint 

### PR DESCRIPTION
**Pull request by HankG**
This is feature complete to the spec with full tests and stimulation through the test harness.

### CODE REVIEW

File: app/controllers/api/v1/postinteractions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class PostinteractionsController < Api::V1::BaseController
```
Comment (frankrousseau) [inline app/controllers/api/v1/postinteractions_controller.rb]: The camelcase doesn't look right.

File: spec/integration/api/post_interactions_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+  describe "#subscribe" do
+    context "succeeds" do
+      it "with proper guid and access token" do
+        post(
+          api_v1_post_subscribe_path(@status.guid),
+          params: {
+            access_token: access_token
+          }
+        )
+        expect(response.status).to eq(204)
```
Comment (frankrousseau) [inline spec/integration/api/post_interactions_controller_spec.rb:30]: You just test that the response status code is right. But you don't test that the post subscription is right. Do you think that this test could be improved?

File: app/controllers/api/v1/postinteractions_controller.rb (hunk 1 above)
Comment (HankG) [reply app/controllers/api/v1/postinteractions_controller.rb]: I agree.  I was having problems with the auto-routing in Ruby when I did PostInteractions.  Let me see if I can fix it.

File: spec/integration/api/post_interactions_controller_spec.rb (hunk 2 above)
Comment (HankG) [reply spec/integration/api/post_interactions_controller_spec.rb:30]: Yes let me see what I can interrogate to show that it properly captured the status change within the test.

File: app/controllers/api/v1/postinteractions_controller.rb (hunk 1 above)
Comment (frankrousseau) [reply app/controllers/api/v1/postinteractions_controller.rb]: Ok don't try too hard. It's not a big issue for me.

File: spec/integration/api/post_interactions_controller_spec.rb (hunk 2 above)
Comment (frankrousseau) [reply spec/integration/api/post_interactions_controller_spec.rb:30]: There is the same problem with tests below.

Comment (HankG) [reply spec/integration/api/post_interactions_controller_spec.rb:30]: All of these tests now have increment checks of the appropriate data structures...

File: app/controllers/api/v1/postinteractions_controller.rb (hunk 1 above)
Comment (HankG) [reply app/controllers/api/v1/postinteractions_controller.rb]: I was able to get this working.  I've gotten better at this as I do the API :)

**Comment by HankG** [comment]
Ready for final review and merging.

File: app/controllers/api/v1/postinteractions_controller.rb (hunk 1 above)
Comment (frankrousseau) [reply app/controllers/api/v1/postinteractions_controller.rb]: Great thank you!

File: spec/integration/api/post_interactions_controller_spec.rb (hunk 2 above)
Comment (frankrousseau) [reply spec/integration/api/post_interactions_controller_spec.rb:30]: Ok thank you

**Pull request merged**